src/python/pants/engine/internals/graph.py

```diff
@@ -151,6 +154,27 @@ def warn_deprecated_field_type(field_type: type[Field]) -> None:
     )
 
 
+@memoized
+def maybe_warn_dependencies_as_copied_field(tgt_type: type[TargetGenerator]) -> None:
```

Comment:
Note that 2.11 already made a breaking change with target generators and moved vs copied fields. Didn't exist in 2.10. I don't think it's necessary to deprecate this particular change? Bundle it with the overall change.

Reply:
This will convert to an error at some point, and there are outstanding PRs that will probably break. It's cheap enough to keep it as a deprecation.